nuxeo-features/nuxeo-platform-csv-export/src/main/java/org/nuxeo/ecm/platform/csv/export/io/DocumentPropertyCSVWriter.java

```diff
@@ -0,0 +1,173 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.ecm.platform.csv.export.io;
+
+import static org.nuxeo.ecm.core.io.registry.reflect.Instantiations.SINGLETON;
+import static org.nuxeo.ecm.core.io.registry.reflect.Priorities.REFERENCE;
+import static org.nuxeo.ecm.platform.csv.export.io.DocumentModelCSVHelper.getVocabulary;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import org.apache.commons.csv.CSVPrinter;
+import org.nuxeo.common.utils.i18n.I18NUtils;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.model.Property;
+import org.nuxeo.ecm.core.api.model.PropertyNotFoundException;
+import org.nuxeo.ecm.core.api.model.impl.ArrayProperty;
+import org.nuxeo.ecm.core.io.marshallers.csv.AbstractCSVWriter;
+import org.nuxeo.ecm.core.io.registry.reflect.Setup;
+import org.nuxeo.ecm.core.schema.types.ListType;
+import org.nuxeo.ecm.core.schema.types.Type;
+import org.nuxeo.ecm.core.schema.types.primitives.BinaryType;
+import org.nuxeo.ecm.directory.Directory;
+import org.nuxeo.ecm.directory.Session;
+
+/**
+ * @since 10.3
+ */
+@Setup(mode = SINGLETON, priority = REFERENCE)
+public class DocumentPropertyCSVWriter extends AbstractCSVWriter<Property> {
+
+    public static final String LIST_DELIMITER = "\n";
+
+    public static final String LANG_CTX_DATA = "lang";
+
+    public static final String NULL_PROPERTY_LABEL = "null property";
+
+    public DocumentPropertyCSVWriter() {
+        super();
+    }
+
+    @Override
+    protected void write(Property entity, CSVPrinter printer) throws IOException {
+        if (entity.isScalar()) {
+            writeScalarProperty(entity, printer);
+        } else if (entity.isList()) {
+            writeListProperty(entity, printer);
+        } else {
+            writeUnsupported(entity.getType(), printer);
+        }
+    }
+
+    @Override
+    protected void writeHeader(Property entity, CSVPrinter printer) throws IOException {
+        printer.printRecord(entity.getXPath());
+    }
+
+    protected void writeScalarProperty(Property entity, CSVPrinter printer) throws IOException {
+        Object value = entity.getValue();
+        Type type = entity.getType();
+        if (type instanceof BinaryType) {
+            writeUnsupported(type, printer);
+        } else {
+            String valueAsString = null;
+            if (value == null) {
+                printer.print(null);
+            } else {
+                valueAsString = type.encode(value);
+                printer.print(valueAsString);
+            }
+            Directory vocabulary = getVocabulary(type);
```

Comment:
The static import of `getVocabulary` is a bit surprising to the reader. Reading it I expected a local method. Try to use static imports only for well-known static methods or constants whose semantic is clear.